sdk/ml/azure-ai-ml/tests/component/unittests/test_command_component_entity.py

```diff
@@ -453,14 +453,21 @@ def test_primitive_output(self):
     def test_component_code_asset_ignoring_pycache(self) -> None:
         component_yaml = "./tests/test_configs/components/basic_component_code_local_path.yml"
         component = load_component(component_yaml)
-        code_folder = "./tests/test_configs/components/helloworld_components_with_env/"
-        with tempfile.TemporaryDirectory(dir=code_folder) as tmp_dir:
-            # create temp folder and write some files under pycache
-            tmp_pycache_folder = Path(tmp_dir) / "__pycache__"
-            tmp_pycache_folder.mkdir()
-            (tmp_pycache_folder / "a.pyc").touch()
-            (tmp_pycache_folder / "b.pyc").touch()
-            # resolve and check if pycache exists in code asset
+        # create some files/folders expected to ignore
+        pycache = Path("./tests/test_configs/components/helloworld_components_with_env/__pycache__")
+        try:
+            pycache.mkdir()
+            expected_exclude = pycache / "a.pyc"
+            expected_exclude.touch()
+            # resolve and test for ignore_file's is_file_excluded
             with component._resolve_local_code() as code:
-                pycache_path = Path(component_yaml).parent / code.path / Path(tmp_dir).name / "__pycache__"
-                assert not pycache_path.exists()
+                excluded = []
+                for root, _, files in os.walk(code.path):
```

Comment:
shall we verify if ignored folder itself is copied? e.g., an empty `__pycache__` folder.

Reply:
It will impact hash calculation

Reply:
not sure do we need this (also how to do this), as upload logic just keep upload path for files, so I understand the creation of folder is kept by service, and ensuring there isn't file path containing `__pycache__` is enough?

Reply:
```
for root, dirs, files in os.walk(code.path):
   assert "__pycache__" not in dirs
...
```
something like this?